src/Store.js

```diff
@@ -1,38 +1,45 @@
 import AppDispatcher from './AppDispatcher';
+import {EventEmitter2} from 'eventemitter2';
 
-export default class Store {
-  static create(options) {
-
-    let instance = options.state ? new options.state() : new options.collection();
-
-    instance.emitChange = (data) => {
-      instance.trigger('onChange', data);
-    };
-
-    instance.subscribe = (successCallback, errorCallback) => {
-      instance.on('onChange', successCallback);
-      errorCallback && instance.on('onError', errorCallback);
-    };
-
-    instance.unsubscribe = (successCallback, errorCallback) => {
-      successCallback && instance.off('onChange', successCallback);
-      errorCallback && instance.off('onError', errorCallback);
-    };
-
-    instance.emitError = (err) => {
-      instance.trigger('onError', err);
-    };
-
-    let _get = instance.get;
-
-    instance.get = (attr) => {
-      return attr ? _get.call(instance, attr) : instance.toJSON();
-    };
-
-    instance.dispatcher = AppDispatcher;
-
-    instance.dispatchToken = AppDispatcher.register(options.handler.bind(instance));
+export default class Store extends EventEmitter2 {
+  // to be compatible with the existing phrontend
+  static create(...args) {
+    return new Store(...args);
+  }
+  constructor({initialState = {}, handler}) {
+    super();
+    this.state = initialState;
 
-    return instance;
+    this.dispatcher = AppDispatcher;
+    this.dispatchToken = AppDispatcher.register(handler.bind(this));
+  }
+  get(attr) {
+    return attr ? this.state[attr] : this.toJSON();
+  }
+  set(attr, val) {
+    if (typeof attr === 'object') Object.assign(this.state, attr);
+    else this.state[attr] = val;
+  }
+  // to be backward compatible
+  parse(data) {
+    return 'string' === typeof data ? JSON.parse(data) : data;
+  }
+  // to be backward compatible
+  toJSON() {
+    return JSON.parse(JSON.stringify(this.state));
+  }
+  emitChange(...data) {
+    this.emit('change', ...data);
+  }
```

Comment:
The API was specifically made restrictive to allow only argument object `data`. Allowing multiple args might lead to inconsistent uses. I think we should stick to 1-arity